File: Milktea/LeetCode/Code/JavaScript/3623.js
const countTrapezoids = function (points) {
  const countPointInY = new Map();
  for (const [x, y] of points) {
    const key = BigInt(y);
    countPointInY.set(key, (countPointInY.get(key) || 0n) + 1n);
  }
  let prefixSum = 0n;
  let res = 0n;
  const mod = 1000000007n;
  for (const count of countPointInY.values()) {
    const v = (count * (count - 1n) / 2n) % mod;
    res = (res + prefixSum * v) % mod;
    prefixSum = (prefixSum + v) % mod;
  }

  return Number(res);
};

File: Milktea/LeetCode/Code/Python/3623.py
class Solution:
    def countTrapezoids(self, points: List[List[int]]) -> int:
        countPointInY = defaultdict(int)
        for x,y in points:
            countPointInY[y] += 1
        
        prefixSum, res = 0, 0
        mod = 10**9 + 7
        for v in countPointInY.values():
            v = v * (v-1) // 2

            res = (res + prefixSum * v) % mod
            prefixSum = (prefixSum + v) % mod

        return res